### PR TITLE
Added Poetry support

### PR DESCRIPTION
**Pull request by chinmayshah99**
## Description
Added support for poetry. Poetry is a package management tool that helps to setup your environment super quickly. More on it here: https://python-poetry.org/

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+authors = ["Chinmay Shah <chinmayshah3899@gmail.com>", "Vadym Doroshenko <dvadym@google.com>"]
+license = "Apache-2.0"
+
+[tool.poetry.dependencies]
+python = "^3.6"
+apache-beam = "^2.29.0"
+pyspark = "^3.1.1"
+absl-py = "^0.12.0"
+dataclasses = {version = "^0.8", python = "~3.6"}
+numpy = "<1.21"
```
Comment (dvadym) [inline pyproject.toml:14]: Just wondering, why <1.21 version limitation? (which currently cover all current versions). Maybe it's better to specify lower bound (something like more than 1.17 or 1.20).

**Review by dvadym — COMMENTED** [review]
Thanks! I have a question (mostly to understand more about poetry :) )

File: pyproject.toml (hunk 1 above)
Comment (chinmayshah99) [reply pyproject.toml:14]: Numpy has dropped support for python 3,6 starting 1.21.

Comment (dvadym) [reply pyproject.toml:14]: Thanks , it makes sense.

**Pull request merged**